src/clients/c/tb_client.zig

```diff
@@ -75,10 +75,14 @@ pub const Logging = struct {
         comptime format: []const u8,
         args: anytype,
     ) void {
-        // Messages are silently dropped if no logging callback is specified - unless they're warn
-        // or err. The value in having those for debugging is too high to silence them, even until
-        // client libraries catch up and implement a callback handler.
-        const callback = logging.callback orelse {
+        // Debug logs are dropped here unless debug is set, because of the potential penalty in
+        // crossing FFI to drop them.
+        // Other messages are silently dropped if no logging callback is specified - unless they're
+        // warn or err. The value in having those for debugging is too high to silence them, even
+        // until client libraries catch up and implement a callback handler.
+        const callback = if (message_level == .debug and logging.debug)
```

Comment:
meant to say

```suggestion
        const callback = if (message_level == .debug and !logging.debug)
```

perhaps?

Could you also separate out this if to be a distinct statement? Right now, the if is quite busy and feels like it handles separate concerns? 

Maybe avoid the ternary at all?

```zig
if (message_level == .debug and !logging.debug) {
    return;
}

if (logging.callback == null and 
      (message_level == .warn or message_level === .err)) {
  std.log.defaultLog(message_level, scope, format, args);
  return;
}

const callback = logging.callback orelse return;
```

Reply:
Whoops! Good catch.

Yeah agreed, I thought originally there might be a tiny performance benefit from a ternary, but I'm probably wrong and client logging is called way less than replica logging so it's less relevant too.